src/Servers/Kestrel/shared/KnownHeaders.cs

```diff
@@ -1428,7 +1428,16 @@ public bool MoveNext()
                     {(!loop.ClassName.Contains("Trailers") ? $@"_next = _collection._contentLength.HasValue ? {loop.Headers.Length - 1} : -1;" : "_next = -1;")}
                     return true;
                 }}
-            }}
+            }}{(loop.ClassName.Contains("Trailers") ? "" : $@"
+
+            private static int GetNext(long bits, bool hasContentLength)
```

Comment:
Should this have `[MethodImpl(MethodImplOptions.AggressiveInlining)]`

Reply:
You tell me.  When do we apply that?

Reply:
When it shouldn't be a function call 😆 
This function is small enough that we would be fine with the code bloat in favor of no function call overhead.